sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3943,6 +3943,43 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-34421: Resolve temporary functions and views in views with CTEs") {
+    val tempViewName = "temp_view"
+
+    withTempView(tempViewName) {
+      spark.udf.register("temp_func", identity[Int](_))
```

Comment:
Could you use `withUserDefinedFunction`?

Reply:
Sure, fixed in: https://github.com/apache/spark/pull/31550/commits/ce34b5ee88ac066270a9c6ac6c50cd546ba38929